aehmc/proposals.py

```diff
@@ -0,0 +1,159 @@
+from typing import Callable, Tuple
+
+import aesara.tensor as aet
+import numpy as np
+from aesara.tensor.random.utils import RandomStream
+from aesara.tensor.var import TensorVariable
+
+from aehmc.integrators import IntegratorStateType
+
+ProposalStateType = Tuple[
+    IntegratorStateType, TensorVariable, TensorVariable, TensorVariable
+]
+
+
+def proposal_generator(kinetic_energy: Callable, divergence_threshold: float):
+    def init(state):
+        _, p, potential_energy, _ = state
+        energy = potential_energy + kinetic_energy(p)
+        return (state, energy, 0.0, -np.inf)
```

Comment:
Is this missing a test?

Reply:
I don't think this needs a unit test, and it should be covered by the integration tests.

Reply:
Some form of coverage is needed, though.  

For instance, we want something that would alert us to any issues arising when the form of `state` changes, because that unpacking and/or the use of the unpacked items are an easy source of refactoring problems.

Reply:
Turns out this function was not needed, so I removed it.